docs/src/components/InteractivePlayground/useAnimatedSensorPlayground/index.tsx

```diff
@@ -0,0 +1,235 @@
+    qx: quaternion.x,
+    qy: quaternion.y,
+    qz: quaternion.z,
+  };
+}
+
+function formatRotationCode({
+  pitch,
+  roll,
+  yaw,
+  qw,
+  qx,
+  qy,
+  qz,
+}: Omit<ValueRotation, 'interfaceOrientation'>) {
+  return `
+  { 
+    "pitch": ${pitch.toFixed(2)},
+    "roll": ${roll.toFixed(2)},
+    "yaw": ${yaw.toFixed(2)},
+
+    "qw": ${qw.toFixed(2)},
+    "qx": ${qx.toFixed(2)},
+    "qy": ${qy.toFixed(2)},
+    "qz": ${qz.toFixed(2)}
+  }
+  `;
+}
+
+function getGyroscope(dx: number, dy: number, dz: number, dt: number) {
+  const { radX, radY, radZ } = degreeToRadian(dx, dy, dz);
+
+  return {
+    x: (1000 * radX) / dt,
+    y: (1000 * radY) / dt,
+    z: (1000 * radZ) / dt,
+  };
+}
+
+function formatGyroscopeCode({ x, y, z }: { x: number; y: number; z: number }) {
+  return `
+  { 
+    "x": ${x.toFixed(2)},
+    "y": ${y.toFixed(2)},
+    "z": ${z.toFixed(2)}
+  }
+  `;
+}
+
+const initialState = {
+  x: 0,
+  y: 0,
+  z: 0,
+  sensorType: 'ROTATION',
+  measurement: {
+    current: null,
+    previous: null,
+    currentX: null,
+    previousX: null,
+    currentY: null,
+    previousY: null,
+    currentZ: null,
+    previousZ: null,
+  },
+};
+
+export default function useAnimatedSensorPlayground() {
+  const [x, setX] = useState(initialState.x);
+  const [y, setY] = useState(initialState.y);
+  const [z, setZ] = useState(initialState.z);
+  const [sensorType, setSensorType] = useState(initialState.sensorType);
+  const [measurement, setMeasurement] = useState(initialState.measurement);
+
+  const resetOptions = () => {
+    setX(initialState.x);
+    setY(initialState.y);
+    setZ(initialState.z);
+    setMeasurement(initialState.measurement);
+  };
+
+  useEffect(() => {
+    let timer: ReturnType<typeof setTimeout>;
+
+    const loop = () => {
+      const now = performance.now();
+      setMeasurement((prev) => ({
+        current: now,
+        previous: prev.current,
+        currentX: x,
+        previousX: prev.currentX,
+        currentY: y,
+        previousY: prev.currentY,
+        currentZ: z,
+        previousZ: prev.currentZ,
+      }));
+      timer = setTimeout(loop, INTERVAL);
+    };
+    loop();
+
+    return () => {
+      clearTimeout(timer);
+    };
+  }, [x, y, z]);
+
+  const controls = (
+    <>
+      <SelectOption
+        label="Sensor type"
+        value={sensorType}
+        onChange={(option) => setSensorType(option)}
+        options={[
+          'ACCELEROMETER',
+          'GYROSCOPE',
+          'GRAVITY',
+          'MAGNETIC_FIELD',
+          'ROTATION',
+        ]}
+        disabledOptions={['ACCELEROMETER', 'MAGNETIC_FIELD']}
+      />
+      <Range
+        label="X axis angle"
+        min={-180}
+        max={180}
+        step={1}
+        value={x}
+        onChange={setX}
+      />
+      <Range
+        label="Y axis angle"
+        min={-180}
+        max={180}
+        step={1}
+        value={y}
+        onChange={setY}
+      />
+      <Range
+        label="Z axis angle"
```

Comment:
I don't like the fact these are actually called XYZ axes. While it makes sense to give some outside perspective, I think it would be better to describe these as "roll", "pitch" and "yaw" for consistency with actual Rotate sensor. Apart from that, the playground is beatiful 😻

Reply:
While I get that `pitch`, `yaw` and `roll` are technically more correct but...

For me as a developer coming from CSS (rotateX, rotateY, rotateZ) and a Minecraft player XYZ coordinates and rotation on these axes are intuitively understood on the spot while I had to learn what Euler angles are not so long ago.

I would advocate for X, Y, Z axis angles rather than yaw, pitch, and roll as the default choice in the example but that's just a personal preference